captcha-validation-service/src/main/java/io/mosip/captcha/service/CaptchaServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+
+		MultiValueMap<String, String> param = new LinkedMultiValueMap<>();
+		param.add("secret", recaptchaSecret);
+		param.add("response", ((CaptchaRequestDTO) captchaRequest).getCaptchaToken().trim());
+
+		GoogleCaptchaDTO captchaResponse = null;
+
+		try {
+			log.info("In captcha service try block to validate the token request via a google verify site rest call"
+							+ ((CaptchaRequestDTO) captchaRequest).getCaptchaToken() + "  " + recaptchaVerifyUrl);
+			
+			captchaResponse = this.restTemplate.postForObject(recaptchaVerifyUrl, param, GoogleCaptchaDTO.class);
+			if (captchaResponse != null) {
+				log.debug("sessionId", "idType", "id", captchaResponse.toString());
+			}
+		} catch (RestClientException ex) {
+			log.error("In captcha service to validate the token request via a google verify site rest call has failed --->"
+							+ ((CaptchaRequestDTO) captchaRequest).getCaptchaToken() + "  " + recaptchaVerifyUrl + "  "
+							+ ex);
+			if (captchaResponse != null && captchaResponse.getErrorCodes() !=null) {
+			throw new CaptchaException(captchaResponse.getErrorCodes().get(0).getErrorCode(),
+					captchaResponse.getErrorCodes().get(0).getMessage());
+			}
+		}
+
+		if (captchaResponse!=null && captchaResponse.isSuccess()) {
+			log.info("In captcha service token request has been successfully verified --->"
+							+ captchaResponse.isSuccess());
+			mainResponse.setId(mosipcaptchaValidateId);
+			mainResponse.setResponsetime(captchaResponse.getChallengeTs());
+			mainResponse.setVersion(version);
+			CaptchaResposneDTO response = new CaptchaResposneDTO();
+			response.setMessage(CAPTCHA_SUCCESS);
+			response.setSuccess(captchaResponse.isSuccess());
+			mainResponse.setResponse(response);
+		} else {
+			if (captchaResponse != null) {
+				log.error("In captcha service token request has failed --->"
+								+ captchaResponse.isSuccess());
+			}
+			mainResponse.setId(mosipcaptchaValidateId);
+			mainResponse.setResponsetime(getCurrentResponseTime());
+			mainResponse.setVersion(version);
+			mainResponse.setResponse(null);
+			ExceptionJSONInfoDTO error = new ExceptionJSONInfoDTO(CaptchaErrorCode.INVALID_CAPTCHA_CODE.getErrorCode(),
+					CaptchaErrorCode.INVALID_CAPTCHA_CODE.getErrorMessage());
+			List<ExceptionJSONInfoDTO> errorList = new ArrayList<ExceptionJSONInfoDTO>();
+			errorList.add(error);
+			mainResponse.setErrors(errorList);
+
+		}
+		return mainResponse;
+
+	}
+
+	private void validateCaptchaRequest(CaptchaRequestDTO captchaRequest) throws InvalidRequestCaptchaException {
+
+	 if (captchaRequest.getCaptchaToken() == null || captchaRequest.getCaptchaToken().trim().length() == 0) {
+		 	log.debug("sessionId", "idType", "id", captchaRequest.toString());
+			throw new InvalidRequestCaptchaException(CaptchaErrorCode.INVALID_CAPTCHA_REQUEST.getErrorCode(),
+					CaptchaErrorCode.INVALID_CAPTCHA_REQUEST.getErrorMessage());
+		}
+	}
+
+	private String getCurrentResponseTime() {
```

Comment:
As this method is duplicated in more than one place let move it to a util class and reuse it here